gobblin-core/src/main/java/gobblin/publisher/BaseDataPublisher.java

```diff
@@ -162,7 +169,8 @@ protected void publishData(WorkUnitState workUnitState, int branchId, Set<Path>
     } else {
       Path parentDir = publisherOutputDir.getParent();
       // Create the parent directory of the final output directory if it does not exist
-      this.fileSystemByBranches.get(branchId).mkdirs(parentDir);
+      WriterUtils.mkdirsWithRecursivePermission(this.fileSystemByBranches.get(branchId), parentDir,
```

Comment:
You can simply do `WriterUtils.mkdirsWithRecursivePermission(this.fileSystemByBranches.get(branchId), publisherOutputDir,
+          this.permissions.get(branchId));`.